src/ExNavigationReducer.js

```diff
@@ -145,6 +145,28 @@ class ExNavigationReducer {
     return _updateNavigator(state, navigatorUID, NavigationStateUtils.pop(navigatorState));
   }
 
+  static [ActionTypes.POP_N](state, { navigatorUID, n }) {
+    invariant(state.navigators[navigatorUID], 'Navigator does not exist.');
+    const navigatorState = state.navigators[navigatorUID];
+
+    if (navigatorState.index === 0) {
+      return state;
+    }
+
+    if (navigatorState.type === 'slidingTab') {
+      return _updateNavigator(
+        state,
+        navigatorUID,
+        {...navigatorState, index: 0 },
+      );
+    }
+
+    const sliceTo = n > navigatorState.routes.length ? 1 : -n;
```

Comment:
I think this solution seems fine to me -- I think if you use `pop(5)` but only have 3 routes in the stack, you probably want it to just work with 3, but warn you. So we should add a `console.warn` in there if in dev mode.